src/test/java/harness/utils/WorkspaceUtils.java

```diff
@@ -25,10 +32,30 @@ public static String createUserFacingId() {
    */
   public static UFWorkspace createWorkspace(TestUser workspaceCreator)
       throws JsonProcessingException {
+    return createWorkspace(workspaceCreator, Optional.empty());
+  }
+
+  /**
+   * Create a new workspace and register it with Janitor if this test is running in an environment
+   * where Janitor is enabled. Tests must use this method in order to register workspaces with
+   * Janitor, direct calls to `terra workspace create` will potentially leak workspaces.
+   *
+   * @param workspaceCreator The user who owns the workspace. This user will be impersonated to in
+   *     the WSM workspaceDelete request.
+   */
+  public static UFWorkspace createWorkspace(
+      TestUser workspaceCreator, Optional<CloudPlatform> platform) throws JsonProcessingException {
     // `terra workspace create --format=json`
+    List<String> argsList =
+        Stream.of("workspace", "create", "--id=" + createUserFacingId())
+            .collect(Collectors.toList());
+    if (platform.isPresent()) { // defaults to GCP otherwise
+      argsList.add("--platform=" + platform.get());
+    }
+
     UFWorkspace workspace =
         TestCommand.runAndParseCommandExpectSuccess(
-            UFWorkspace.class, "workspace", "create", "--id=" + createUserFacingId());
+            UFWorkspace.class, argsList.toArray(new String[0]));
```

Comment:
nit: just toArray()?

Reply:
list.toArray() will convert the list to Object[], but the parameter is expected to be String[]
Hence this needs to be either cast to a string array or provide destination. 
The latter is preferred to avoid casting